### PR TITLE
[READY] Port hair gradient customization

### PR DESCRIPTION
**Pull request by dearmochi**
**This is a loose port of https://github.com/tgstation/tgstation/pull/59435**

# ~~TODO~~
- [x] appearance editor support
- [x] preferences UI support
- [x] ensure it works right for all species
- [x] ensure SQL is fine

## What Does This PR Do

- Adds the option to overlay hairstyles with a gradient of varying offset, color and transparency.
  - It is a character preference and can be changed in the mirror like normal hair.

## Why It's Good For The Game

- Hair styles can be quite tricky to work with as they are right now, so gradients can help fill the gap and allow for fancy effects.

## Images of changes

dreamseeker_8aljTlEiZo dreamseeker_p6fduGPDZw

## Changelog
:cl:
add: Add hair gradient customization in Preferences and mirrors
/:cl:

### CODE REVIEW

File: SQL/paradise_schema_prefixed.sql [hunk 1]
```diff
@@ -276,13 +280,13 @@ CREATE TABLE `SS13_player` (
...
+  `keybindings` longtext COLLATE utf8mb4_unicode_ci DEFAULT NULL,
+  `keybindings` longtext COLLATE utf8mb4_unicode_ci DEFAULT NULL,
   `2fa_status` ENUM('DISABLED','ENABLED_IP','ENABLED_ALWAYS') NOT NULL DEFAULT 'DISABLED' COLLATE 'utf8mb4_general_ci',
   PRIMARY KEY (`id`),
   UNIQUE KEY `ckey` (`ckey`),
   KEY `lastseen` (`lastseen`),
-  KEY `computerid` (`computerid`),
-  KEY `ip` (`ip`),
-  KEY `fuid` (`fuid`),
+  KEY `co
```
Comment (dearmochi) [inline SQL/paradise_schema_prefixed.sql:289]: what????

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 2]
```diff
@@ -354,49 +354,60 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
 		else
 			//warning("Invalid ha_style for [species.name]: [ha_style]")
 
-
-
-//HAIR OVERLAY
+/**
+  * Generates overlays for the hair layer.
+  */
 /mob/living/carbon/human/proc/update_hair()
```
Comment (dearmochi) [inline code/modules/mob/living/carbon/human/update_icons.dm:360]: Completely rewrote this proc to not use icon ops, in favour of overlays. Icon ops are generally expensive and disk dependent. This should serve as the basis for a complete update icons refactor later.

**Comment by AffectedArc07** [comment]
StatTrak Bedhead Long - Fade

File: SQL/paradise_schema_prefixed.sql (hunk 1 above)
Comment (AffectedArc07) [inline SQL/paradise_schema_prefixed.sql:289]: wack

File: SQL/paradise_schema_prefixed.sql [hunk 3]
```diff
@@ -276,13 +280,13 @@ CREATE TABLE `SS13_player` (
   `fupdate` smallint(4) DEFAULT '0',
   `parallax` tinyint(1) DEFAULT '8',
   `byond_date` DATE DEFAULT NULL,
+  `keybindings` longtext COLLATE utf8mb4_unicode_ci DEFAULT NULL,
+  `keybindings` longtext COLLATE utf8mb4_unicode_ci DEFAULT NULL,
```
Comment (AffectedArc07) [inline SQL/paradise_schema_prefixed.sql:284]: wat

File: code/modules/mob/new_player/preferences_setup.dm [hunk 4]
```diff
@@ -346,6 +346,19 @@
 		else if(hair_style.do_colouration)
 			hair_s.Blend(h_colour, ICON_ADD)
 
+		// mochi TODO: rewrite the mess that is this proc
```
Comment (dearmochi) [inline code/modules/mob/new_player/preferences_setup.dm:349]: I mean it, this is atrocious

**Review by SteelSlayer — APPROVED** [review]
LGTM

**Review by AffectedArc07 — APPROVED** [review]
SQL changes work

**Pull request merged**